lint-consul-retry.go

```diff
@@ -230,20 +244,34 @@ func walkDir(path string) error {
 
 func visitFile(path string, f os.FileInfo, err error) error {
 	if err != nil {
-		return fmt.Errorf("failed to visit '%s', %v", err)
+		return fmt.Errorf("failed to visit '%s', %v", path, err)
 	}
-	if isTestFile(path, f) {
-		tree, _ := parser.ParseFile(fset, path, nil, parser.ParseComments)
 
-		// Only process files importing sdk/testutil/retry
-		if importsRetry(tree) {
-			v := visitor{}
-			ast.Walk(v, tree)
-		}
+	if !isGoTestFile(path, f) {
+		return nil
 	}
+
+	tree, err := parser.ParseFile(fset, path, nil, parser.ParseComments)
+	if err != nil {
+		return fmt.Errorf("failed to parse test file '%s', %v", path, err)
+	}
+
+	// Only process files importing sdk/testutil/retry
+	if importsRetry(tree) {
+		v := visitor{path: path}
+		ast.Walk(v, tree)
+	}
+
 	return nil
 }
 
-func isTestFile(path string, f os.FileInfo) bool {
-	return !f.IsDir() && strings.Contains(path, "test")
+func isGoFile(path string, f os.FileInfo) bool {
+	if !f.Mode().IsRegular() {
+		return false
+	}
+	return strings.HasSuffix(path, ".go")
```

Comment:
I added this condition to avoid having it process files that were completely irrelevant.